src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -57,28 +61,62 @@ private void InstrumentModule()
                 {
                     foreach (var type in module.GetTypes())
                     {
-                        if (type.CustomAttributes.Any(a => a.AttributeType.Name == "ExcludeFromCoverageAttribute" || a.AttributeType.Name == "ExcludeFromCoverage"))
-                            continue;
-
-                        foreach (var method in type.Methods)
-                        {
-	                        var sourceFile = method.DebugInformation.SequencePoints.Select(s => s.Document.Url).FirstOrDefault();
-	                        if (!string.IsNullOrEmpty(sourceFile) && _excludedFiles.Contains(sourceFile)) {
-	                            continue;
-	                        }
-                            if (!method.CustomAttributes.Any(a => a.AttributeType.Name == "ExcludeFromCoverageAttribute" || a.AttributeType.Name == "ExcludeFromCoverage"))
-                                InstrumentMethod(method);
-                        }
+                        InstrumentType(type);
                     }
 
                     module.Write(stream);
                 }
             }
         }
 
+        private void InstrumentType(TypeDefinition type)
+        {
+            if (type.CustomAttributes.Any(IsExcludeAttribute))
+                return;
+
+            foreach (var method in type.Methods)
+            {
+                if (!method.CustomAttributes.Any(IsExcludeAttribute))
+                    InstrumentMethod(method);
+            }
+
+            foreach (var property in type.Properties)
+            {
+                if (!property.CustomAttributes.Any(IsExcludeAttribute))
+                    InstrumentProperty(property);
+            }
+
+            if (type.HasNestedTypes)
+            {
+                foreach (var nestedType in type.NestedTypes)
+                    InstrumentType(nestedType);
```

Comment:
Did you confirm that this actually fixes the specified issues. It was my thinking that Cecil returned every type regardless of whether they were nested or not

Reply:
When I was adding the branch logic, I saw something in OpenCover's symbol manager: https://github.com/OpenCover/opencover/blob/35f1299e136c9f91241839b85e20d7eb6ed24416/main/OpenCover.Framework/Symbols/CecilSymbolManager.cs#L183

Reply:
Sure. I've just changed the logic slightly to use `module.Types` instead of `module.GetTypes()` so that this logic for nested types can be retained without processing nested types more than once.

Reply:
@sjp - Actually, I think @tonerdo is correct.  `GetTypes()` from `Mono.Cecil.Rocks` already handled nested types.  He was also correct that Properties were created as methods as seen in the attached screenshot (all that went up was hit count when properties were added).

I actually would kind of prefer letting Cecil handle type nesting.

<img width="1164" alt="slack - red ventures 2018-05-01 16-10-30" src="https://user-images.githubusercontent.com/747670/39491598-eabcb944-4d5a-11e8-900b-d2a8c2c3d54a.png">